src/www/app/app.ts

```diff
@@ -408,6 +414,9 @@ export class App {
       throw new Error(`disconnectServer event had no server ID`);
     }
 
+    if (!this.canServerConnectionChange(serverId)) return;
```

Comment:
*minor* I feel we can move this `if` statement to `setServerConnectionChangeTimeout` to reduce the duplicated code here.

Reply:
We'd also still need to handle the value returned from `setServerConnectionChangeTimeout` and return from this function anyway to short circuit it, so I'm not sure what that would DRY up, if I'm understanding you correctly.

Are you suggesting something like this?

```js
const didThrottle = this.throttleServerConnectionChange(serverId, DEFAULT_SERVER_CONNECTION_STATUS_CHANGE_TIMEOUT);


if (didThrottle) return;
```

Reply:
Ah true, that could be come a single line - one moment